src/frontend/components/PopularMenu.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from 'react'
+
+export default class PopularMenu extends Component {
+    render() {
```

Comment:
This rendering is just too big. Consider breaking it down into simple components.

Reply:
Do you want me to break individual `list` into components?